### PR TITLE
cast logits to half when T=MLFloat16

### PR DESCRIPTION
**Pull request by wangyems**

### CODE REVIEW

**Comment by tianleiwu** [comment]
Please add a test case for regression test.

**Review by yufenglee — APPROVED** [review]
:shipit:
